### DISCUSSION TITLE
Solax Inverter and EV Charger RS485 TO ETH

### DISCUSSION THREAD

**Discussion by arikatame** [wills106/homeassistant-solax-modbus#1045] [Q&A] [answered]
Hello all,

I have set up the Inverter X3-Hybrid-G4 with the integration and it is working perfectly fine. I also have a Solax EV Charger. 

My Question is: Is it possible to get the values from the EV Charger as well over the same Modbus? 
I see the Values in the Solax app but since it only refreshes every 5 minutes I would like to rebuild the live energy diagram in HA.
Therefore I would need a sensor for the current Power from the EV Charger. My Inverter doesn't show that so I guess I would get that from the EV Charger.

I use: [Waveshare RS485 to RJ45 Ethernet DIN Rail Mount - RS485 TO POE ETH (B)] connected to the Modbus between Inverter and EV Charger.

I'm not sure what I need to set up in the Waveshare adapter to get both Inverter and EV Charger. Also how do I find the Modbus address from the Charger? My Inverter works with address 1, when I try to set up a second device on the same adaptor with address 2 it just keeps loading.
 
Any help would be appreciated .:)

**Comment by wills106** [comment, marked as answer]
Have a look at #349 and https://github.com/PatrikTrestik/homeassistant-solax-http

If the charger is set to have the Inverter as the data source only the WiFi will give data, so you need to use the above.

To communicate to the charger with Modbus you need to either use CT Clamps as the data source, but you loose Green mode.
Or you can look at using https://github.com/PatrikTrestik/homeassistant-solax-evc-proxy

**Reply by arikatame** [reply]
Thanks for the helpful links. 
I got to connect to the EVC via Homeassistant SolaX Http.
Only little issue I had connecting was, that I had to use the registration number to connect instead of the mentioned serial number.
I didn't had time to test everything yet but I'm positive that it will work.
